cli_tools/osconfig_agent/inventory_test.go

```diff
@@ -0,0 +1,139 @@
+//  Copyright 2017 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/go/packages"
+)
+
+func decodePackages(str string) packages.Packages {
+	decoded, _ := base64.StdEncoding.DecodeString(str)
+	zr, _ := gzip.NewReader(bytes.NewReader(decoded))
+	var buf bytes.Buffer
+	io.Copy(&buf, zr)
+	zr.Close()
+
+	var pkgs packages.Packages
+	json.Unmarshal(buf.Bytes(), &pkgs)
+	return pkgs
+}
+
+func TestWriteInventory(t *testing.T) {
```

Comment:
Do we have the ability to test an unhappy case (check the Error []string field) yet?

Reply:
The Error field is going to be removed in favor of reporting errors via some other method, and then we will have to add a test for the various failure modes